LayoutTests/imported/w3c/web-platform-tests/css/selectors/user-invalid.html

```diff
@@ -12,6 +32,31 @@
   assert_false(input.matches(':user-invalid'));
 }, ':user-invalid selector should be supported');
 
+promise_test(async () => {
+  const input = document.querySelector("#user-invalid-test");
+  assert_false(input.matches(":user-valid"), "Initially does not match :user-valid");
+  assert_false(input.matches(":user-invalid"), "Initially does not match :user-invalid");
+
+  await test_driver.click(input);
+
+  input.blur();
+
+  assert_false(input.matches(":user-valid"), "No change happened, still does not match :user-valid");
+  assert_false(input.matches(":user-invalid"), "No change happened, still does not match :user-invalid");
+
+  await test_driver.click(input);
+
+  await test_driver.send_keys(input, "not an email");
+  input.blur();
+  assert_true(input.matches(":user-invalid"), "Typed an invalid email, :user-invalid now matches");
+  assert_false(input.matches(":user-valid"), "Typed an invalid email, :user-valid does not match");
+
+
+  input.value = "";
+  assert_true(input.matches(":user-valid"), "Put a valid email, :user-valid now matches");
+  assert_false(input.matches(":user-invalid"), "Put an valid email, :user-invalid no longer matches");
```

Comment:
We just cleared the input here, not enter a valid email.

Reply:
Added `input.required = true`.